subsys/net/ip/6lo.c

```diff
@@ -756,7 +756,7 @@ static inline bool compress_IPHC_header(struct net_pkt *pkt,
 
 	/* UDP header compression */
 	if (IS_ENABLED(CONFIG_NET_UDP)) {
-		struct net_udp_hdr hdr, *udp;
+		struct net_udp_hdr hdr __unused, *udp;
```

Comment:
You are right, was confused by IS_ENABLED but checking my own config, NET_UDP was not even used. Will change as suggested.